RedirectApplication/RedirectApplication/Json/Json.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace RedirectApplication.Json
+{
+    public class JsonWorking
+    {
+        public void Main(string tree)
+        {
+            //var tree = System.IO.File.ReadAllText("tree.json");
+            var reader = new JsonTextReader(new StringReader(tree));
+            PostJson content = JsonSerializer.CreateDefault().Deserialize<PostJson>(reader);
+
+            //File.WriteAllText("rules.json",JsonConvert.SerializeObject(content));
+        }
+    }
+
+    class TreeNodeConverter : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            serializer.Serialize(writer, value);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var array = JArray.Load(reader);
+            var target = new List<ITreeNode>();
+
+
+            if (array.HasValues)
+            {
+                foreach (var child in array.Children())
+                {
+                    var node = CreateNode(child);
+
+                    serializer.Populate(child.CreateReader(), node);
+                    target.Add(node);
+                }
+            }
+            return target;
+        }
+
+        private ITreeNode CreateNode(JToken obj)
+        {
+            var type = (string)obj["name"];
+
+            switch (type)
```

Comment:
Replace switch with `Dictionary<string, Func<ITreeNode>>`, each dictionary value is a factory function for a corresponding class. So you can just pick needed factory by string, call it and return the result. If there is no such entry - throw an exception